sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2028,4 +2028,18 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       Row(false) :: Row(true) :: Nil)
   }
 
+  test("SPARK-12340: overstep the bounds of Int in SparkPlan.executeTake"){
+    val rdd = sqlContext.sparkContext.parallelize(1 to 3 , 3 )
+    
+    rdd.toDF("key").registerTempTable("spark12340")
+    checkAnswer(
+      sql("select key from spark12340 limit 2147483638"),    
+      Row(1) :: Row(2) :: Row(3) :: Nil
+    )
+    
+    assert(rdd.take(2147483638).size === 3)
+    
+    assert(rdd.takeAsync(2147483638).get.size === 3)
```

Comment:
Does this test case pass the compilation? `2147483638` is not within the range of `Int`.
Should we say `Int.MaxValue` ?